src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -110,47 +133,333 @@ class boss_gluth : public CreatureScript
+    };
+
+};
+
+// spell 28374 (10man) / 54426 (25man) -  Decimate
+class spell_gluth_decimate : public SpellScriptLoader
+{
+public:
+    spell_gluth_decimate() : SpellScriptLoader("spell_gluth_decimate") { }
+
+    class spell_gluth_decimate_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gluth_decimate_SpellScript);
 
-            if (me->GetVictim() && me->EnsureVictim()->GetEntry() == NPC_ZOMBIE)
+        // This method handles the damaging effect of the decimate spell.
+        void HandleScriptEffect(SpellEffIndex /* index */)
+        {
+            if (Unit *unit = GetHitUnit())
             {
-                if (me->IsWithinMeleeRange(me->GetVictim()))
+                if (unit->GetEntry() == NPC_ZOMBIE || unit->GetTypeId() == TypeID::TYPEID_PLAYER || unit->IsPet())
                 {
-                    me->Kill(me->GetVictim());
-                    me->ModifyHealth(int32(me->CountPctFromMaxHealth(5)));
+                    int32 damage = int32(unit->GetHealth()) - int32(unit->CountPctFromMaxHealth(5));
+                    if (damage > 0)
+                        GetOriginalCaster()->CastCustomSpell(SPELL_DECIMATE_DMG, SPELLVALUE_BASE_POINT0, damage, unit);
                 }
             }
+        }
+
+        // This method handles the change of the zombie's behavior after a decimate.
+        void HandleSendEvent(SpellEffIndex /* index */)
+        {
+            if (GetOriginalCaster()->GetEntry() == NPC_GLUTH)
+                GetOriginalCaster()->ToCreature()->GetAI()->DoAction(DO_ZOMBIE_RUN);
+        }
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (sSpellMgr->GetSpellInfo(SPELL_DECIMATE_DMG))
+                return true;
+            else
+                return false;
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_gluth_decimate_SpellScript::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+            OnEffectHit += SpellEffectFn(spell_gluth_decimate_SpellScript::HandleSendEvent, EFFECT_2, SPELL_EFFECT_SEND_EVENT);
+        }
+    };
+
+    SpellScript* GetSpellScript() const
+    {
+        return new spell_gluth_decimate_SpellScript();
+    }
+
+};
+
+class spell_gluth_zombiechow_search : public SpellScriptLoader
+{
+public:
+    spell_gluth_zombiechow_search() : SpellScriptLoader("spell_gluth_zombiechow_search") { }
+
+    class spell_gluth_zombiechow_search_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gluth_zombiechow_search_SpellScript);
+
+        void HealForEachTargetHit()
+        {
+            GetOriginalCaster()->ModifyHealth(int32(GetOriginalCaster()->CountPctFromMaxHealth(5)));
+        }
+        
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (sSpellMgr->GetSpellInfo(SPELL_ZOMBIE_CHOW_SEARCH_MULTI) && sSpellMgr->GetSpellInfo(SPELL_ZOMBIE_CHOW_SEARCH_SINGLE))
```

Comment:
Valid should be removed actually. Both spell will call this spell script. If this spell script is called, this automaticly mean the spell that called it exists in the db. No need to check for the existence of another spell. I misused Validate() on that one.